app/interactors/users/save.rb

```diff
@@ -0,0 +1,19 @@
+module Users
+  class Save
+    include Interactor
+
+    delegate :user_params, to: :context
+
+    def call
+      context.user = user
+
+      context.fail! unless user.update(user_params)
+    end
+
+    private
+
+    def user
+      @user ||= context.user || User.new
+    end
+  end
+end
```

Comment:
```suggestion
module Users
  class Save
    include Interactor

    delegate :user, :user_params, to: :context
    
    before do
      context.user ||= User.new
    end

    def call
      context.fail!("Invalid data") unless user.update(user_params)
    end
  end
end
```

Reply:
not working with these changes: "message": "undefined method `each' for \"Invalid data\":String"
save.rb:10:in `call'",

Reply:
I updated my comment